src/server/index.jsx

```diff
@@ -288,19 +290,41 @@ server
         headers,
       });
 
+      let derivedPageType = 'Unknown';
+
       try {
-        const { service, isAmp, route, variant } = getRouteProps(
-          routes,
-          urlPath,
-        );
-        const data = await route.getInitialData({
+        const {
+          service,
+          isAmp,
+          route: { getInitialData, pageType },
+          variant,
+        } = getRouteProps(routes, urlPath);
+
+        // Set derivedPageType based on matched route
+        derivedPageType = pageType || derivedPageType;
+
+        const data = await getInitialData({
           path: url,
           service,
           variant,
         });
+
         const { status } = data;
         const bbcOrigin = headers['bbc-origin'];
 
+        // Set derivedPageType based on returned page data
+        if (status === 200) {
+          derivedPageType = ramdaPath([('pageData', 'metadata', 'type')], data);
```

Comment:
Just a query, are we combining two different things here, the pageType as identified by the routing and the type from the page data?

Reply:
Not combining, we have a situation where we can match a route and get the generic pageType e.g. `cpsAsset` then if we then successfully fetch the page data we can set the pageType to what is declared in the data i.e. STY or MAP.

If we fail to fetch the data for whatever reason we just use the default value of the value from the matched route

We have the default set to Unknown incase we do not match a route or fetch the page data,